dts/bindings/gpio/adi,max22190-gpio.yaml

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2010-2024 Analog Devices Inc.
+# Copyright (c) 2024 BayLibre SAS
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  ADI MAX22190 octal industrial Input with advanced diagnostic
+  capabilities like Wire break and over/under volatage.
+  Fiter configuration could be done on per channel bases, which
+  mean WB functionality and fiters could be set.Reffering to:
+  filter-wbes = <CH0 CH1 CH2 ... CH7 >  for wire break
+  To ebale wire break set 1 else 0. Same principal is followed for:
+  filter-fbps and filter-dealys.
+  Sample binding
+  &sdp_spi {
+    status = "okay";
+    pinctrl-names = "default";
+    max22190_gpio0: max22190_gpio@0 {
+      compatible = "adi,max22190-gpio";
+      reg = <0>;
+
+      spi-max-frequency = <1000000>;
+      status = "okay";
+
+      gpio-controller;
+      #gpio-cells = <2>;
+      ngpios = <8>;
+      status = "okay";
+
+      max22190-mode = <1>; // modes range from 0-4
+
+      drdy-gpios = <&gpioj 12 GPIO_ACTIVE_LOW>; /* SDP-GPIO5 - PMOD-PIN8 */
+      fault-gpios = <&gpioc 11 GPIO_ACTIVE_LOW>; /* SDP-SERIAL_INT - PMOD-PIN7 */
+      latch-gpios = <&gpioj 13 GPIO_ACTIVE_LOW>; /* SDP-GPIO6 - PMOD-PIN9 */
+
+      filter-wbes = <1 0 0 0  0 0 0 0>; // wirebreak WBEx
+      filter-fbps = <0 0 0 0  0 0 0 0>; // programmable filter INx
+      filter-delays = <50 100 400 800 1600 3200 12800 20000>; // 1000 us == 1ms
+    };
+  };
+
+compatible: "adi,max22190-gpio"
+
+properties:
+  "#gpio-cells":
+    const: 2
+  ngpios:
+    type: int
+    required: true
+    const: 8
+    description: Number of gpios supported
+  drdy-gpios:
+    description: Ready pin which show when chip is ready
+    type: phandle-array
+  fault-gpios:
+    description: |
+      Fault pin indicates when there is Fault state in either FAULT1 or FAULT2
+      bothe of which are cleaned on read once problem is not persistent
+    type: phandle-array
+  latch-gpios:
+    description: |
+      Latch the data so it could be read (partially duplicate CS)
+    type: phandle-array
+  max22190-mode:
+    type: int
+    default: 1
```

Comment:
default value requires justification in description (same for other properties)

Reply:
sure, I add clarification for all. Could you take a look if it is enough ?

Reply:
I specified default value because this was mode set on the devkit it self, but it make sens to have it in DT.